taxonomy/signals/signals.py

```diff
@@ -8,3 +8,5 @@
 UPDATE_COURSE_SKILLS = Signal()
 UPDATE_PROGRAM_SKILLS = Signal()
 UPDATE_XBLOCK_SKILLS = Signal()
+XBLOCK_DELETED = Signal()
+XBLOCK_DUPLICATED = Signal()
```

Comment:
I am unsure about these signals for the reason I [mentioned in the course discovery PR](https://github.com/openedx/course-discovery/pull/3710#pullrequestreview-1201958033) and also because, these signals cannot be actually emitted by the taxonomy connector for other apps to listen.

What the taxonomy connector could possibly emit are `DELETE_XBLOCK_SKILLS`, `DUPLICATE_XBLOCK_SKILLS` at the start of the processing or  `XBLOCK_SKILLS_DELETED`, `XBLOCK_SKILLS_DUPLICATED` at the end of the processing depending on the kind of signal that's emitted.

Again, it seems the only reason for these signals to be defined here, is to allow for them to be handled in `handlers.py`. I would suggest we think if this is actually necessary. Wouldn't it be better to not define them at all and do the following in handlers.py?

```diff
from .signals import (
    UPDATE_COURSE_SKILLS,
    UPDATE_PROGRAM_SKILLS,
    UPDATE_XBLOCK_SKILLS,
-    XBLOCK_DELETED,
-    XBLOCK_DUPLICATED,
)
+ from openedx_events.content_authoring.signals import XBLOCK_DELETED, XBLOCK_DUPLICATED, XBLOCK_PUBLISHED
```

I guess the answer again is in the question, why can't we add *openedx-events* as a dependency to the *taxonomy-connector*.

Reply:
@tecoholic I don't have a strong reason to not include openedx-events as a dependency to taxonomy-connector. `openedx-events` is already added as a dependency in course-discovery through [event-bus-kafka](https://github.com/openedx/event-bus-kafka) so I did not include it here. This also keeps taxonomy-connector independent of `openedx-events` if that matters at all. I think we can ask @sameenfatima78 for her advice here.

Reply:
Discussed this with @saleem-latif and while we both agree with using standardized signals as suggested by @tecoholic, we are still unsure about how `taxonomy-connector` would listen these events if these signals are not being fired from `course-discovery`? But other than that, installing this new dependency in `taxonomy-connector` won't be an issue.

Reply:
@sameenfatima78 

>we are still unsure about how taxonomy-connector would listen these events if these signals are not being fired from course-discovery

In the current design, they are fired from `course-discovery` as implemented in https://github.com/openedx/course-discovery/pull/3710. But if we directly include `openedx-events` as dependency in `taxonomy`, it is not required. The events fired in https://github.com/openedx/edx-platform/pull/31350 can be directly handled here.

Reply:
@tecoholic @sameenfatima78 Updated to include `openedx-events` directly as dependency and handle xblock signals. Please review.